Delete endpoint

lizchow1 opened the pull request:
Changes:
- initially added the new columns into the database ie minutes, tips etc
- added delete user by firebase ID into the backend, tested using curl requests
- connected the delete endpoint into the settings page so that the user can delete their account
- added a modal that is used to confirm that a user wants to delete their account and then to re enter their password before deleting
- changed  seed_dummy_data() to call on https://dummyjson.com/docs/recipes to seed 100 recipes, tested using curl -X POST http://127.0.0.1:5000/api/recipes/seed
- centered the recipe card on the homepage and moved the no recipes text to below the title on the homepage

Screenshot showing the centered recipe card:
![Screenshot 2025-03-29 at 12 03 29](https://github.com/user-attachments/assets/01fec84d-4ff4-4745-99bd-2b6bef231a22)

Screenshot showing the setting page:
![Screenshot 2025-03-29 at 12 03 46](https://github.com/user-attachments/assets/9937bd02-cc62-4788-a5b7-07a6c6c3d2f6)

Screenshot showing the confirm deletion modal:
![Screenshot 2025-03-29 at 12 03 59](https://github.com/user-attachments/assets/3876388b-93e6-4b68-9549-47fea1700dad)

Screenshot showing the enter password to delete account modal:
![Screenshot 2025-03-29 at 12 04 18](https://github.com/user-attachments/assets/46db8776-5eb8-432c-ba8f-77469ab20ff4)

lizchow1 committed:
add new columns to database

lizchow1 committed:
centre recipe card on homepage

lizchow1 committed:
add delete user endpoint

lizchow1 committed:
connect delete api

lizchow1 committed:
added delete modal

lizchow1 committed:
move no recipes text

lizchow1 committed:
use dummy recipe json api

viktoriabjorklund reviewed (commented):
Looks good. However, some redundancy right now since you can change password both in settings and profile, maybe it's better do leave that functionality in just settings. So maybe just remove the UserDetailsForm in UserProfilePage

lizchow1 commented:
> Looks good. However, some redundancy right now since you can change password both in settings and profile, maybe it's better do leave that functionality in just settings. So maybe just remove the UserDetailsForm in UserProfilePage

I have removed it in the friends branch and replaced it there, I can remove it here too though

lizchow1 committed:
remove edit details form
